internal/extender/resource.go

```diff
@@ -429,6 +528,38 @@ func (s *SparkSchedulerExtender) rescheduleExecutor(ctx context.Context, executo
 	}
 	executorResources := &resources.Resources{CPU: sparkResources.executorResources.CPU, Memory: sparkResources.executorResources.Memory, NvidiaGPU: sparkResources.executorResources.NvidiaGPU}
 	availableNodes := s.getNodes(ctx, nodeNames)
+
+	shouldScheduleIntoSingleAZ := false
+	singleAzZone := ""
+	if doesBinpackingScheduleInSingleAz(s.binpacker) && s.shouldScheduleDynamicallyAllocatedExecutorsInSameAZ {
+		svc1log.FromContext(ctx).Info("Dynamic Allocation single AZ scheduling enabled, attempting to get zone to schedule into.")
+		zone, allPodsInSameAz, err := s.getCommonZoneForExecutorsApplication(ctx, executor)
+		if err != nil {
+			return "", "", err
+		}
+		if allPodsInSameAz {
+			svc1log.FromContext(ctx).Info("Only considering nodes from the zone",
+				svc1log.SafeParam("zone", zone))
+
+			availableNodes, err = filterNodesToZone(ctx, availableNodes, zone)
```

Comment:
I suppose we're assuming that if got here, `availableNodes` is going to be >0? Can we add a comment to clarify?

Reply:
Yep, we'd throw in https://github.com/palantir/k8s-spark-scheduler/blob/1524603df98c80b53a6d71a1e534f399517fbacc/internal/extender/resource.go#L537

I'll add a comment